src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -2394,11 +2394,17 @@ static int __icap_xclbin_download(struct icap *icap, struct axlf *xclbin)
 			goto out;
 		}
 	}
-
-
-	err = icap_download_bitstream(icap, xclbin);
-	if (err)
-		goto out;
+// dont need to program the bitstream for flat shell
+	if(xclbin->m_header.m_mode != XCLBIN_FLAT)
+	{
+    err = icap_download_bitstream(icap, xclbin);
+    if (err)
+      goto out;
+  }
+  else
```

Comment:
For flat shell case, if someone is trying to download an xclbin here, don't we want to return error back if the uuid of the xclbin is different than what is flashed on the board? Or how do we know if the xclbin passed down matches the one we have on the board?

Reply:
Hi @maxzhen , Yes , we should add that check. I will add it.

Reply:
added the logic 
 ->get uuid from board and compare with xclbin uuid 
 if both are same no need to program the board otherwise return with error